lib/health_checking/health_checks/issue_commentable.rb

```diff
@@ -0,0 +1,48 @@
+module HealthChecks
+  class IssueCommentableCheck
+    include HealthCheck
+
+    name 'issue_comment_check'
+    weight :info
+    authorization :all
+    message 'Login if you want to comment on this boards issues'
+
+    ## deps
+    # {
+    #   repo: repo object,
+    #   authorization: :all, :collaborator or :admin,
+    #   logged_in: bool
+    # }
+    ##
+    
+    def perform(deps)
+      return not_authorized unless authorized?(deps)
+      return pass if deps[:logged_in]
+      return failure
```

Comment:
I think perform has too many responsibilities, it's doing a check and building a payload.

It seems like what you are going for is a DSL above to configure the desired payload

`name`, `weight`, `message` contains all you need to build a consistent payload. 

perform should be able to return `true` or `false`

lets assume perform **WONT** be called if there isn't sufficient authorization, at that point its comes down to a `pass` or `fail`